### PR TITLE
Feature/celery refactor

### PR DESCRIPTION
**Pull request by rkuo-danswer**
## Description
Fixes DAN-821.

## How Has This Been Tested?
[Describe the tests you ran to verify your changes]

## Accepted Risk
[Any know risks or failure modes to point out to reviewers]

## Related Issue(s)
[If applicable, link to the issue(s) this PR addresses]

## Checklist:
- [ ] All of the automated tests pass
- [ ] All PR comments are addressed and marked resolved
- [ ] If there are migrations, they have been rebased to latest main
- [ ] If there are new dependencies, they are added to the requirements
- [ ] If there are new environment variables, they are added to all of the deployment methods
- [ ] If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- [ ] Docker images build and basic functionalities work
- [ ] Author has done a final read through of the PR right before merge

### CODE REVIEW

File: backend/supervisord.conf [hunk 1]
```diff
@@ -33,9 +33,9 @@ stopasgroup=true
 # since this is often the bottleneck for "sync" jobs (e.g. document set syncing, 
 # user group syncing, deletion, etc.) 
 [program:celery_worker_light]
-command=bash -c "celery -A danswer.background.celery.celery_run:celery_app worker \
+command=bash -c "celery -A danswer.background.celery.versioned_apps.light worker \
     --pool=threads \
-    --concurrency=${CELERY_WORKER_LIGHT_CONCURRENCY:-24} \
+    --concurrency=${CELERY_WORKER_LIGHT_CONCURRENCY:-${NUM_INDEXING_WORKERS:-24}} \
```
Comment (rkuo-danswer) [inline backend/supervisord.conf]: wrong var, fix

File: backend/danswer/background/celery/versioned_apps/heavy.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+"""Factory stub for running celery worker / celery beat."""
+from celery import Celery
+
+from danswer.utils.variable_functionality import set_is_ee_based_on_env_variable
+
+set_is_ee_based_on_env_variable()
+
+
+def get_app() -> Celery:
```
Comment (Weves) [inline backend/danswer/background/celery/versioned_apps/heavy.py:11]: small: can we add a comment as to why these are different from `beat.py`? And a comment about how we need to make a change to these to match `beat.py` if we want to add `ee` specific tasks?

File: backend/supervisord.conf (hunk 1 above)
Comment (Weves) [inline backend/supervisord.conf]: small merge issue, `NUM_INDEXING_WORKERS` should not be here.

Also, can we move all of these configs over to the config files directly?

**Review by Weves — CHANGES_REQUESTED** [review]
Can we remove the `celery_run.py` file, since it's unused?

**Review by Weves — APPROVED** [review]

**Pull request merged**